public/app/plugins/datasource/testdata/StreamHandler.ts

```diff
@@ -0,0 +1,395 @@
+    super(key, query, request, observer);
+    window.setTimeout(() => {
+      this.stream.series = [this.initBuffer(query.refId)];
+      this.looper();
+    }, 10);
+  }
+
+  nextRow = (time: number) => {
+    const { spread, noise } = this.query;
+    this.value += (Math.random() - 0.5) * spread;
+    return [
+      time,
+      this.value, // Value
+      this.value - Math.random() * noise, // MIN
+      this.value + Math.random() * noise, // MAX
+    ];
+  };
+
+  initBuffer(refId: string): SeriesData {
+    const { speed, buffer } = this.query;
+    const data = {
+      fields: [
+        { name: 'Time', type: FieldType.time },
+        { name: 'Value', type: FieldType.number },
+        { name: 'Min', type: FieldType.number },
+        { name: 'Max', type: FieldType.number },
+      ],
+      rows: [],
+      refId,
+      name: 'Signal ' + refId,
+    } as SeriesData;
+
+    const request = this.stream.request;
+    data.meta = {
+      request: request.requestId,
+    };
+
+    this.value = Math.random() * 100;
+    const maxRows = buffer ? buffer : request.maxDataPoints;
+    let time = Date.now() - maxRows * speed;
+    for (let i = 0; i < maxRows; i++) {
+      data.rows.push(this.nextRow(time));
+      time += speed;
+    }
+    return data;
+  }
+
+  looper = () => {
+    if (!this.observer) {
+      const request = this.stream.request;
+      const elapsed = request.startTime - Date.now();
+      if (elapsed > 1000) {
+        console.log('Stop looping');
+        return;
+      }
+    }
+
+    // Make sure it has a minimum speed
+    const { query } = this;
+    if (query.speed < 5) {
+      query.speed = 5;
+    }
+
+    this.appendRows([this.nextRow(Date.now())]);
+    this.timeoutId = window.setTimeout(this.looper, query.speed);
+  };
+}
+
+export class LogsWorker extends StreamWorker {
+  index = 0;
+
+  constructor(key: string, query: TestDataQuery, request: DataQueryRequest, observer: DataStreamObserver) {
+    super(key, query, request, observer);
+
+    window.setTimeout(() => {
+      this.stream.series = [this.initBuffer(query.refId)];
+      this.looper();
+    }, 10);
+  }
+
+  getNextWord() {
+    this.index = (this.index + Math.floor(Math.random() * 5)) % words.length;
+    return words[this.index];
+  }
+
+  getRandomLine() {
+    let line = this.getNextWord();
+    while (line.length < 80) {
+      line += ' ' + this.getNextWord();
+    }
+    return line;
+  }
+
+  nextRow = (time: number) => {
+    return [time, this.getRandomLine()];
+  };
+
+  initBuffer(refId: string): SeriesData {
+    const { speed, buffer } = this.query;
+    const data = {
+      fields: [{ name: 'Time', type: FieldType.time }, { name: 'Line', type: FieldType.string }],
+      rows: [],
+      refId,
+      name: 'Logs ' + refId,
+    } as SeriesData;
+
+    const request = this.stream.request;
+    data.meta = {
+      request: request.requestId,
```

Comment:
can't find this used anywhere, there is a requestId prop on QueryResultMeta as well

Reply:
Good catch -- i just removed it.  This is an artifact of maintaining #16676

I'll revist this later -- the meta.requestId (not request!) gets used to pick the right sub-request to show in the QueryEditor -- it shows thing like how long it has been running etc.